### ISSUE TITLE
E-mail addresses are compared case sensitive

### ISSUE THREAD

**Issue by mhier** [saltcorn/saltcorn#1167]
**Describe the bug**
Saltcorn expects the e-mail address on login to be type in the exact case as stored in the user table. This confuses frequently users if they type their mail address with a different casing. In case of the "Forgot password?" feature, it won't even display an error message if an unknown mail address is entered, but no mail will be sent.

**To Reproduce**
1. Create user with mixed case mail address
2. Log out
3. Try to log in with the new user but type the mail address all-lowercase
4. Get error: "Incorrect user or password"

then also:

5. Click "Forgot password?" link
6. Enter same mail address all-lowercase (or any other non-existent mail address)
7. See message "Email with password reset link sent", but no mail sent.

**Expected behavior**
1. Either use always case insensitive comparison when searching for users by mail address, or
2. add admin option whether mail addresses shall be case sensitive or not.

also:

3. Give error message when an unknown mail address is entered in the "Forgot password" form.

**Screenshots**
(I can provide screen shots, but I think they wouldn't really help here.)

**Additional context**
To my knowledge, mail addresses are technically case sensitive before the @ sign (domain names are certainly not case sensitive), but all mail providers consider mail addresses case insensitive. It would be very confusing if a mail provider would direct mail addresses only differing in the casing to different people. Hence I would consider it safe to consider mail addresses always as case insensitive. In case not everyone agrees with me here I suggest to introduce this as an admin option (although I think that makes it just unnecessarily complicated to implement).

**Comment by glutamate** [comment]
I have been postponing this because of the security concerns around Unicode characters (see https://dev.to/jagracey/hacking-github-s-auth-with-unicode-s-turkish-dotless-i-460n) but I can look for a secure solution

**Comment by mhier** [comment]
Good point. I am not really an expert in this field, but I could imagine that first converting the string to punycode, then to lowercase for the comparison, in addition to using always the mail address stored in the data base for email sending and for display, would fix the security issues while keeping backwards compatibility for existing setups.

**Comment by Greg-Malski** [comment]
Just curious if any additional headway has been made/considered on this? Perhaps an admin checkbox to enable/disable case sensitivity?
